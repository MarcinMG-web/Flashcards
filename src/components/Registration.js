import { is } from '@babel/types';
import React, { useState, useRef } from 'react';
import { Link } from 'react-router-dom';
import { register } from '../../src/services/Register';
import { isUser, isEmail } from '../services/Authorization';

const Registration = () => {
  const [registrationParams, setRegistrationParams] = useState({
    email: '',
    password: '',
    username: '',
  });

  const initialError = {
    errorEmail: '',
    errorPassword: '',
    errorUsername: '',
  };
  const [errorsValidation, setErrorsValidation] = useState(initialError);

  // iRegistration info
  const [sendFormText, setSendFormText] = useState('');
  const [showRegistrationInfo, setShowRegistrationInfo] = useState(false);

  const formRef = useRef();

  const handleChange = (e) => {
    setRegistrationParams({
      ...registrationParams,
      [e.target.name]: e.target.value,
    });
  };

  const checkEmail = async (email) => {
    const isEmailResponse = await isEmail(email);
    const isValidEmail = await isEmailResponse['data'][email];
    return isValidEmail;
  };

  const checkUser = async (username) => {
    const isUserResponse = await isUser(username);
    const isValidUser = await isUserResponse['data'][username];
    return isValidUser;
  };

  const validate_email = async () => {
    const isEmailChecked = await checkEmail(registrationParams.email);
    if (isEmailChecked === 'True') {
      alert('This email has been already used')
    };
    return isEmailChecked;
  };

  const validate_username = async () => {
    const isUserChecked = await checkUser(registrationParams.username);
    if (isUserChecked === 'True') {
      alert('This username is already in use')
    };
    return isUserChecked;
  };

  // validate_email(registrationParams.email);
  // validate_username(registrationParams.username);

  const validate = async () => {
    let errorEmail = '';
    let errorUsername = '';
    let errorPassword = '';
    validate_email();
    validate_username();

    // const isEmailChecked = await checkEmail(registrationParams.email);
    // console.log(isEmailChecked);
    // const isUserChecked = await checkUser(registrationParams.username);
    // console.log(isUserChecked);

    if (registrationParams.email.length === 0) {
      errorEmail = 'set the email';
    }

    // if (isEmailChecked === 'True') {
    //   alert('This email has been already used')
    //   errorEmail = 'This email has been already used';
    // }

    if (registrationParams.username.length === 0) {
      errorUsername = 'set the user name';
    }

    // if (isUserChecked === 'True') {
    //   errorUsername = 'This username is already in use';
    // }

    if (registrationParams.password.length === 0) {
      errorPassword = 'set the password';
    }

    if (errorEmail ||  errorUsername || errorPassword) {
      setErrorsValidation({
        errorEmail,
        errorUsername,
        errorPassword,
      });
      return false;
    }
    return true;
  };

  const check = () => {
    const isValid = validate();

    if (isValid) {
      /**
       * Clear Error
       */
      setErrorsValidation(initialError);
    }
  };

  const isRegistrationToApplication = (e) => {
    e.preventDefault();
    const isValid = validate();

    if (isValid) {
      /**
       * Registration
       */
      register(registrationParams);
      setShowRegistrationInfo(true);
      setSendFormText(
        'Thank you for registering in FlipCards and choosing our product. Your account was successfully created and is almost ready to use. Open your email, click the link below to verify your email so we make sure everything is up and running!'
      );
    }

    if (formRef.current) {
      formRef.current.reset();
    }
  };

  return (
    <div className='ContainerRegistration'>
      <div className='ContainerRegistration_header'>Registration Panel</div>

      <div className='ContainerRegistration_main'>
        <div className='ContainerRegistration_main-form'>
          {showRegistrationInfo ? (
            <div>
              {sendFormText}
              <Link to='./' type='submit' className='btn_start-afterRegister'>
                After that go to login panel
              </Link>
            </div>
          ) : (
            <form onSubmit={isRegistrationToApplication} ref={formRef}>
              <div className='ContainerLogin_main-form-input'>
                <label style={{ margin: '15px' }}>Enter your email:</label>
                <input
                  type='email'
                  placeholder='user@domain.com'
                  name='email'
                  onChange={handleChange}
                  onBlur={check}
                  className='inputLogin'
                />
              </div>

              <span style={{ color: 'red', fontSize: '14px' }}>
                {errorsValidation.errorEmail}
              </span>

              <div className='ContainerRegistration_main-form-input'>
                <label style={{ margin: '15px' }}>Enter your user name:</label>
                <input
                  type='text'
                  placeholder='username'
                  name='username'
                  onChange={handleChange}
                  onBlur={check}
                  className='inputLogin'
                />
              </div>

              <span style={{ color: 'red', fontSize: '14px' }}>
                {errorsValidation.errorUsername}
              </span>

              <div className='ContainerRegistration_main-form-input'>
                <label style={{ margin: '15px' }}>Set your password:</label>
                <input
                  type='password'
                  placeholder='password'
                  name='password'
                  onChange={handleChange}
                  onBlur={check}
                  className='inputLogin'
                />

                <span style={{ color: 'red', fontSize: '14px' }}>
                  {errorsValidation.errorPassword}
                </span>

                <input className='special' name='field_name' type='text' />

                <div className='ContainerRegistration_main-form-button'>
                  <Link to='./' type='submit' className='btn_start'>
                    Back to the login panel
                  </Link>

                  <button type='submit' className='btn_createAccount'>
                    Create account
                  </button>
                </div>
              </div>
            </form>
          )}
        </div>
      </div>
      <div className='ContainerRegistration_footer'></div>
    </div>
  );
};

export default Registration;
