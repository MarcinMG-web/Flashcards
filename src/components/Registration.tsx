import React, { useState } from 'react';

import { Link } from 'react-router-dom';
import { register } from '../services/Register';
import { isUser, isEmail } from '../services/Authorization';
import { AxiosResponse } from 'axios';

type InitialRegistration = {
  email: string;
  username: string;
  password: string;
  repeatPassword: string;
};

type InitialError = {
  errorEmail?: string;
  errorUsername?: string;
  errorPassword?: string;
  errorRepeatPassword?: string;
};

const Registration: React.FC = (): JSX.Element => {
  const [registrationParams, setRegistrationParams] =
    useState<InitialRegistration>({
      email: '',
      username: '',
      password: '',
      repeatPassword: '',
    });

  /**
   * Validation input errors
   */
  const initialError = {
    errorEmail: '',
    errorUsername: '',
    errorPassword: '',
    errorRepeatPassword: '',
  };
  const [errorsValidation, setErrorsValidation] =
    useState<InitialError>(initialError);

  /**
   * Registration info
   */
  const [sendFormText, setSendFormText] = useState<string>('');
  const [showRegistrationInfo, setShowRegistrationInfo] =
    useState<boolean>(false);

  const handleChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setRegistrationParams({
      ...registrationParams,
      [e.target.name]: e.target.value,
    });
  };

  const checkEmail = async (email: string): Promise<boolean> => {
    const isEmailResponse: AxiosResponse | any = await isEmail(email);
    const isValidEmail = await isEmailResponse['data'][email];
    return isValidEmail;
  };

  const checkUser = async (username: string): Promise<boolean> => {
    const isUserResponse: AxiosResponse | any = await isUser(username);
    const isValidUser: boolean = await isUserResponse['data'][username];
    console.log('isValidUser', isValidUser);
    return isValidUser;
  };

  const validateEmail = async (): Promise<boolean> => {
    let errorEmail: string = '';
    const isEmailChecked = await checkEmail(registrationParams.email);
    if (isEmailChecked) {
      errorEmail = 'This email has been already used';
      setErrorsValidation({
        errorEmail,
      });
    } else {
      errorEmail = '';
      setErrorsValidation({
        errorEmail,
      });
    }
    return isEmailChecked;
  };

  const validateUsername = async (): Promise<boolean> => {
    let errorUsername: string = '';
    const isUserChecked = await checkUser(registrationParams.username);
    if (isUserChecked) {
      errorUsername = 'This username is already in use';
      setErrorsValidation({
        errorUsername,
      });
    } else {
      errorUsername = '';
      setErrorsValidation({
        errorUsername,
      });
    }
    return isUserChecked;
  };

  const validate = (): boolean => {
    let errorEmail: string = '';
    let errorUsername: string = '';
    let errorPassword: string = '';
    let errorRepeatPassword: string = '';

    validateEmail();
    validateUsername();

    if (registrationParams.email.length === 0) {
      errorEmail = 'set the email';
    }

    if (registrationParams.username.length === 0) {
      errorUsername = 'set the user name';
    }

    if (registrationParams.password.length === 0) {
      errorPassword = 'set the password';
    }

    if (registrationParams.repeatPassword.length === 0) {
      errorRepeatPassword = 'set the repeat password';
    }

    if (registrationParams.password !== registrationParams.repeatPassword) {
      (errorPassword = 'set the same password') ||
        (errorRepeatPassword = 'set the same password');
    }

    if (errorEmail || errorUsername || errorPassword || errorRepeatPassword) {
      setErrorsValidation({
        errorEmail,
        errorUsername,
        errorPassword,
        errorRepeatPassword,
      });
      return false;
    }
    return true;
  };

  const check = (): void => {
    const isValid: boolean = validate();

    if (isValid) {
      /**
       * Clear Error
       */
      setErrorsValidation(initialError);
    }
  };

  const isRegistrationToApplication = (
    e: React.SyntheticEvent<EventTarget>
  ) => {
    e.preventDefault();
    const isValid: boolean = validate();

    if (isValid) {
      /**
       * Registration
       */
      register(registrationParams);
      setShowRegistrationInfo(true);
      setSendFormText(
        'Thank you for registering in FlipCards and choosing our product. Your account was successfully created and is almost ready to use. Open your email, click the link below to verify your email so we make sure everything is up and running!'
      );
    }
  };

  return (
    <div className='ContainerRegistration'>
      <div className='ContainerRegistration_header'>Registration Panel</div>

      <div className='ContainerRegistration_main'>
        <div className='ContainerRegistration_main-form'>
          {showRegistrationInfo ? (
            <div>
              {sendFormText}
              <Link to='./' type='submit' className='btn_start-afterRegister'>
                After that go to login panel
              </Link>
            </div>
          ) : (
            <form onSubmit={isRegistrationToApplication}>
              <div className='ContainerLogin_main-form-input'>
                <label style={{ margin: '15px' }}>Enter your email:</label>
                <input
                  type='email'
                  placeholder='user@domain.com'
                  name='email'
                  onChange={handleChange}
                  onBlur={check}
                  className='inputLogin'
                />
              </div>

              <span style={{ color: 'red', fontSize: '14px' }}>
                {errorsValidation.errorEmail}
              </span>

              <div className='ContainerRegistration_main-form-input'>
                <label style={{ margin: '15px' }}>Enter your user name:</label>
                <input
                  type='text'
                  placeholder='username'
                  name='username'
                  onChange={handleChange}
                  onBlur={check}
                  className='inputLogin'
                />
              </div>

              <span style={{ color: 'red', fontSize: '14px' }}>
                {errorsValidation.errorUsername}
              </span>

              <div className='ContainerRegistration_main-form-input'>
                <label style={{ margin: '15px' }}>Set your password:</label>
                <input
                  type='password'
                  placeholder='password'
                  name='password'
                  onChange={handleChange}
                  onBlur={check}
                  className='inputLogin'
                />

                <span style={{ color: 'red', fontSize: '14px' }}>
                  {errorsValidation.errorPassword}
                </span>
              </div>

              <div className='ContainerRegistration_main-form-input'>
                <label style={{ margin: '15px' }}>
                  Set repeated your password:
                </label>
                <input
                  type='password'
                  placeholder='repeated your password'
                  name='repeatPassword'
                  onChange={handleChange}
                  onBlur={check}
                  className='inputLogin'
                />

                <span style={{ color: 'red', fontSize: '14px' }}>
                  {errorsValidation.errorRepeatPassword}
                </span>
              </div>

              <div className='ContainerRegistration_main-form-button'>
                <Link to='./' type='submit' className='btn_start'>
                  Back to the login panel
                </Link>

                <button type='submit' className='btn_createAccount'>
                  Create account
                </button>
              </div>
            </form>
          )}
        </div>
      </div>
      <div className='ContainerRegistration_footer'></div>
    </div>
  );
};

export default Registration;
