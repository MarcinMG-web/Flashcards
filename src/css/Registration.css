.ContainerRegistration {
  width: 600px;
  height: 100vh;
  min-width: 300px;
  background-color: #fff;
  color: #000;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
}

.ContainerRegistration_header {
  width: 100%;
  height: 60px;
  background-color: #000;
  color: #fff;
  display: flex;
  justify-content: center;
  align-items: center;
}

.ContainerRegistration_main {
  display: flex;
  flex-direction: column;
}

.ContainerRegistration_main-form {
  width: 35vw;
}

.ContainerRegistration_main-form-button {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  margin-top: 10px;
}

.ContainerRegistration_footer {
  width: 100%;
  height: 60px;
  background-color: #fff;
  display: flex;
  justify-content: center;
  align-items: center;
}

.inputLogin {
  width: 100%;
  height: 40px;
  border: 1px solid orange;
  border-radius: 5px;
  text-align: center;
  padding: 10px;
  margin-top: 10px;
  margin-bottom: 10px;
  font-family: 'New Tegomin', serif;
  font-size: 16px;
}

@media (max-width: 466px) {
  .ContainerRegistration {
    width: 100vw;
    height: 100vh;
    margin: 0;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
  }
  .ContainerRegistration_header {
    width: 100%;
    height: 60px;
    background-color: #000;
    color: #fff;
    display: flex;
    justify-content: center;
    align-items: center;
  }
  .ContainerRegistration_main {
    display: flex;
    flex-direction: column;
  }
  .ContainerRegistration_main-form {
    width: 90vw;
  }
  .ContainerRegistration_main-form-button {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    margin-top: 10px;
  }
  .ContainerRegistration_footer {
    width: 100%;
    height: 60px;
    background-color: #fff;
    display: flex;
    justify-content: center;
    align-items: center;
  }
}

@media (min-width: 465px) and (max-width: 768px) {
  .ContainerRegistration {
    width: 100vw;
    height: 100vh;
    margin: 0;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
  }
  .ContainerRegistration_header {
    width: 100%;
    height: 60px;
    background-color: #000;
    color: #fff;
    display: flex;
    justify-content: center;
    align-items: center;
  }
  .ContainerRegistration_main {
    display: flex;
    flex-direction: column;
  }
  .ContainerRegistration_main-form {
    width: 90vw;
  }
  .ContainerRegistration_main-form-button {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    margin-top: 10px;
  }
  .ContainerRegistration_footer {
    width: 100%;
    height: 60px;
    background-color: #fff;
    display: flex;
    justify-content: center;
    align-items: center;
  }
}
