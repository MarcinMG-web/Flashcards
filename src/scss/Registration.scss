.ContainerRegistration {
  width: 600px;
  height: 100vh;
  min-width: 300px;
  background-color: #fff;
  color: #000;

  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;

  &_header {
    width: 100%;
    height: 60px;
    background-color: #000;
    color: #fff;
    display: flex;
    justify-content: center;
    align-items: center;
  }
  &_main {
    display: flex;
    flex-direction: column;
    &-form {
      width: 35vw;
      &-button {
        display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: center;
        margin-top: 10px;
      }
    }
  }

  &_footer {
    width: 100%;
    height: 60px;
    background-color: black;
    display: flex;
    justify-content: center;
    align-items: center;
  }
}

.inputLogin {
  width: 100%;
  height: 40px;
  border: 1px solid rgb(255, 165, 0);
  border-radius: 5px;

  text-align: center;
  padding: 10px;
  margin-top: 10px;
  margin-bottom: 10px;
  font-family: 'New Tegomin', serif;
  font-size: 16px;
}
