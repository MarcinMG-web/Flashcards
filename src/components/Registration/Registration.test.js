import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import Registration from './Registration';

import { BrowserRouter } from 'react-router-dom';

const MockRegistration = () => {
  return (
    <BrowserRouter>
      <Registration />
    </BrowserRouter>
  );
};

describe('Registration Tests - render', () => {
  it('snapshot', () => {
    const tree = render(<MockRegistration />);

    expect(tree).toMatchSnapshot();
  });

  it('render without crashing', () => {
    render(<MockRegistration />);
  });
});

describe('Tests Business Logic', () => {
  it('show validation info when click create account in empty form', async () => {
    render(<MockRegistration />);

    const buttonCreateAccount = screen.getByText(/Create account/i);
    const validationInfoEmail = 'set the email';
    const validationInfoUserName = 'set the user name';
    const validationInfoPassword = 'set the password';
    const validationInfoRepeatedPassword = 'set the repeat password';

    fireEvent.click(buttonCreateAccount);

    await waitFor(() => {
      const emailMassageValidation = screen.getByTestId(
        'span-validationEmail-message'
      );

      const userNameMassageValidation = screen.getByTestId(
        'span-validationUsername-message'
      );

      const passwordMassageValidation = screen.getByTestId(
        'span-validationPassword-message'
      );

       const repetendPasswordMassageValidation = screen.getByTestId(
         'span-validationRepeatPassword-message'
       );

      expect(emailMassageValidation).toHaveTextContent(validationInfoEmail);
      expect(userNameMassageValidation).toHaveTextContent(
        validationInfoUserName
      );
      expect(passwordMassageValidation).toHaveTextContent(
        validationInfoPassword
      );

      expect(repetendPasswordMassageValidation).toHaveTextContent(
        validationInfoRepeatedPassword
      );

    });
  });
});
