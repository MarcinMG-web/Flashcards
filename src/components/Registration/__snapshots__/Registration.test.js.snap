// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration Tests - render snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="ContainerRegistration"
      >
        <div
          class="ContainerRegistration_header"
        >
          Registration Panel
        </div>
        <div
          class="ContainerRegistration_main"
        >
          <div
            class="ContainerRegistration_main-form"
          >
            <form>
              <div
                class="ContainerLogin_main-form-input"
              >
                <label
                  style="margin: 15px;"
                >
                  Enter your email:
                </label>
                <input
                  class="inputLogin"
                  name="email"
                  placeholder="user@domain.com"
                  type="email"
                />
              </div>
              <span
                data-testid="span-validationEmail-message"
                style="color: red; font-size: 14px;"
              />
              <div
                class="ContainerRegistration_main-form-input"
              >
                <label
                  style="margin: 15px;"
                >
                  Enter your user name:
                </label>
                <input
                  class="inputLogin"
                  name="username"
                  placeholder="username"
                  type="text"
                />
              </div>
              <span
                data-testid="span-validationUsername-message"
                style="color: red; font-size: 14px;"
              />
              <div
                class="ContainerRegistration_main-form-input"
              >
                <label
                  style="margin: 15px;"
                >
                  Set your password:
                </label>
                <input
                  class="inputLogin"
                  name="password"
                  placeholder="password"
                  type="password"
                />
                <span
                  data-testid="span-validationPassword-message"
                  style="color: red; font-size: 14px;"
                />
              </div>
              <div
                class="ContainerRegistration_main-form-input"
              >
                <label
                  style="margin: 15px;"
                >
                  Set repeated your password:
                </label>
                <input
                  class="inputLogin"
                  name="repeatPassword"
                  placeholder="repeated your password"
                  type="password"
                />
                <span
                  data-testid="span-validationRepeatPassword-message"
                  style="color: red; font-size: 14px;"
                />
              </div>
              <div
                class="ContainerRegistration_main-form-button"
              >
                <a
                  class="btn_start"
                  href="/"
                  type="submit"
                >
                  Back to the login panel
                </a>
                <button
                  class="btn_createAccount"
                  type="submit"
                >
                  Create account
                </button>
              </div>
            </form>
          </div>
        </div>
        <div
          class="ContainerRegistration_footer"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="ContainerRegistration"
    >
      <div
        class="ContainerRegistration_header"
      >
        Registration Panel
      </div>
      <div
        class="ContainerRegistration_main"
      >
        <div
          class="ContainerRegistration_main-form"
        >
          <form>
            <div
              class="ContainerLogin_main-form-input"
            >
              <label
                style="margin: 15px;"
              >
                Enter your email:
              </label>
              <input
                class="inputLogin"
                name="email"
                placeholder="user@domain.com"
                type="email"
              />
            </div>
            <span
              data-testid="span-validationEmail-message"
              style="color: red; font-size: 14px;"
            />
            <div
              class="ContainerRegistration_main-form-input"
            >
              <label
                style="margin: 15px;"
              >
                Enter your user name:
              </label>
              <input
                class="inputLogin"
                name="username"
                placeholder="username"
                type="text"
              />
            </div>
            <span
              data-testid="span-validationUsername-message"
              style="color: red; font-size: 14px;"
            />
            <div
              class="ContainerRegistration_main-form-input"
            >
              <label
                style="margin: 15px;"
              >
                Set your password:
              </label>
              <input
                class="inputLogin"
                name="password"
                placeholder="password"
                type="password"
              />
              <span
                data-testid="span-validationPassword-message"
                style="color: red; font-size: 14px;"
              />
            </div>
            <div
              class="ContainerRegistration_main-form-input"
            >
              <label
                style="margin: 15px;"
              >
                Set repeated your password:
              </label>
              <input
                class="inputLogin"
                name="repeatPassword"
                placeholder="repeated your password"
                type="password"
              />
              <span
                data-testid="span-validationRepeatPassword-message"
                style="color: red; font-size: 14px;"
              />
            </div>
            <div
              class="ContainerRegistration_main-form-button"
            >
              <a
                class="btn_start"
                href="/"
                type="submit"
              >
                Back to the login panel
              </a>
              <button
                class="btn_createAccount"
                type="submit"
              >
                Create account
              </button>
            </div>
          </form>
        </div>
      </div>
      <div
        class="ContainerRegistration_footer"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
